design/baremetal-operator/early-cleaning.md

```diff
@@ -0,0 +1,125 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Early out-of-band cleaning
+
+This document proposes running driver-specific out-of-band clean
+steps.
+
+## Status
+
+implementable
+
+## Summary
+
+Depending on the selected driver, BMO will run zero or more out-of-band clean
+steps to ensure the node and the BMC are in the right state. Out-of-band clean
+steps are executed via BMC (e.g. using Redfish protocol) without booting the
+ironic agent (IPA).
+
+The first drivers to receive such handling will be `idrac` and
+`idrac-virtualmedia`. For them we will run the `management.known_good_state`
+clean step that resets iDRAC and clears its job queue.
+
+It is important that these actions run before we try to boot anything on the
+target node, be it the inspection ramdisk or a 3rd party installer ISO.
+
+## Motivation
+
+iDRAC will refuse to create a job if there is already a pending job of the same
+type. To make sure we start in a coherent state, we will reset iDRAC before
+doing anything else to it.
+
+### Goals
+
+* Provide a mechanism to run out-of-band clean steps before inspection.
+* Allow a user to opt out of this mechanism on per node basis.
```

Comment:
Why?

Reply:
A safety precaution in case of a crazy BMC? Let's drop it for now.